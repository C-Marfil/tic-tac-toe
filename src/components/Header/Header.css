

.header-title {
    text-align: center;
    color: var(--secondary-color);
    font-size: 50px;
  }