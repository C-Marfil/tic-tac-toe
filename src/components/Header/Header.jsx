import React from "react";
import { useNavigate } from "react-router-dom";
import "./Header.css";

const Header = () => {
  const navigate = useNavigate();

  const handleHome = () => {
    navigate("/");
  };

  return (
    <header>
      <h1 className="header-title">Tic-Tac-Toe</h1>
      <button type="button" onClick={handleHome}>
        Home
      </button>
    </header>
  );
};

export default Header;
