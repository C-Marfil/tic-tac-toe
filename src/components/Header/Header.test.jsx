import React from "react";
import { render, screen } from "@testing-library/react";
import { BrowserRouter, MemoryRouter } from "react-router-dom";
import Header from "./Header";

describe("Header Component, children and Routing render/work properly", () => {
  const validProps = {
    setUsername: jest.fn(),
    username: "Carlos",
  };

  it("Snapshot test", () => {
    const { asFragment } = render(
      <Header
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Tests the header renders", () => {
    render(
      <Header
        setUsername={validProps.setUsername}
        username={validProps.username}
      />,
      { wrapper: BrowserRouter }
    );

    expect(screen.getByAltText(/logo/i).toBeInTheDocument);
  });

  it("Tests the navigation works", () => {
    const route = "/lobby";

    render(
      <MemoryRouter initialEntries={[route]}>
        <Header
          setUsername={validProps.setUsername}
          username={validProps.username}
        />
      </MemoryRouter>
    );
    expect(screen.getByAltText(/logo/i).toBeInTheDocument);
  });
});
