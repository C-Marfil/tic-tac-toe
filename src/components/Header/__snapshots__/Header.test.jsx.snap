// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component, children and Routing render/work properly Snapshot test 1`] = `
<DocumentFragment>
  <header>
    <h1
      class="header-title"
    >
      Connect4 To Be
    </h1>
    <button
      type="button"
    >
      Home
    </button>
  </header>
</DocumentFragment>
`;
